cdap-app-fabric/src/test/java/co/cask/cdap/metadata/FieldLineageAdminTest.java

```diff
@@ -72,11 +89,69 @@ public void testFields() {
                         fieldLineageAdmin.getFields(endPoint, 0, Long.MAX_VALUE, "add", false));
   }
 
+  @Test
+  public void testFieldsWithDsSchema() throws Exception {
+    FieldLineageAdmin fieldLineageAdmin = new FieldLineageAdmin(new FakeFieldLineageReader(getFieldNames(),
+                                                                                           Collections.emptySet(),
+                                                                                           Collections.emptySet()),
+                                                                metadataAdmin);
+    EndPoint endPoint = EndPoint.of(NamespaceId.DEFAULT.getNamespace(), "file");
+
+    // test that when there is no schema information present for the dataset and the we request for lineage with
+    // includeCurrent set to true we get lineage fields correctly.
+    Set<Field> expected = getFields(getFieldNames());
+    // includeCurrent set to true
+    Set<Field> actual = fieldLineageAdmin.getFields(endPoint, 0, Long.MAX_VALUE, null, true);
+    Assert.assertEquals(expected, actual);
+
+    // schema with fields which are different than known to lineage store
+    Schema schema =
+      Schema.recordOf("record",
+                      Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                      Schema.Field.of("address", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                      Schema.Field.of("addiffField1", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                      Schema.Field.of("diffField2", Schema.nullableOf(Schema.of(Schema.Type.INT)))
+      );
+
+    // add the the dataset with the schema with fields known in lineage store
+    TableProperties.Builder props = TableProperties.builder();
+    TableProperties.setSchema(props, schema);
+    TableProperties.setRowFieldName(props, "name");
+    datasetFramework.addInstance("table", new DatasetId(NamespaceId.DEFAULT.getNamespace(), "file"), props.build());
+
+    // test all fields expected should have all the fields which was known the lineage store but should not contains
+    // any dataset schema field since the includeCurrent is set to false
+    expected = getFields(getFieldNames());
+    actual = fieldLineageAdmin.getFields(endPoint, 0, Long.MAX_VALUE, null, false);
+    Assert.assertEquals(expected, actual);
+
+    // test all fields expected should have all the fields which was known the lineage store and also the fields
+    // which were only present in the dataset schema since includeCurrent is set to true.
+    // this also test that for the fields which are common in lineage store and dataset schema for example address in
+    // this case has their lineage info field set to true as we do have lineage for this field
+    expected = getFields(getFieldNames());
+    expected.addAll(new HashSet<>(Arrays.asList(new Field("addiffField1", false), new Field("diffField2", false))));
```

Comment:
nit: I think it's more readable to just add the fields individually on two lines.